lib/iex/lib/iex/evaluator.ex

```diff
@@ -81,31 +81,54 @@ defmodule IEx.Evaluator do
     )
   end
 
-  def parse(input, opts, {buffer, last_op}) do
+  def parse(input, opts, {buffer, last_op, adjusted_op}) do
     input = buffer ++ input
     file = Keyword.get(opts, :file, "nofile")
     line = Keyword.get(opts, :line, 1)
     column = Keyword.get(opts, :column, 1)
 
     result =
       with {:ok, tokens} <- :elixir.string_to_tokens(input, line, column, file, opts),
-           {:ok, adjusted_tokens} <- adjust_operator(tokens, line, column, file, opts, last_op),
+           {:ok, adjusted_tokens, adjusted_op} <-
+             adjust_operator(tokens, line, column, file, opts, last_op),
            {:ok, forms} <- :elixir.tokens_to_quoted(adjusted_tokens, file, opts) do
         last_op =
           case forms do
             {:=, _, [_, _]} -> :match
             _ -> :other
           end
 
+        forms =
+          if adjusted_op do
+            quote do
+              case Process.get(:iex_error) do
+                {_kind, _error, _stacktrace} ->
+                  IO.write(:stdio, [
+                    "Skippping evaluation of:\n\n",
+                    unquote(input),
+                    "\n\n",
+                    "For safety, you cannot begin an expression with `#{unquote(adjusted_op)}` when the last expression was an error.\n"
+                  ])
+
+                  IEx.dont_display_result()
+
+                _ ->
+                  unquote(forms)
+              end
+            end
+          else
+            forms
+          end
```

Comment:
I think we need to continue raising... otherwise the pipe after the failed pipe would work:

```
start
|> invalid([)
|> good() #=> this will be aborted
|> good() #=> this will run because we cleared the state
```

So we need something like this:

```suggestion
        forms =
          if adjusted_op != nil and Process.get(:iex_error) != nil do
            quote do
             reraise RuntimeError.exception("skipping evaluation of expression because pipeline has failed"), []
            end
          else
            forms
          end
```

Reply:
Ah, it is a separate process, you are right. Parsing happens elsewhere. I will drop another suggestion then!